phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/PagingRegionScanner.java

```diff
@@ -72,15 +96,10 @@ private boolean next(List<Cell> results, boolean raw) throws IOException {
                 // There is no more row from the HBase region scanner. We need to check if PageFilter
                 // has stopped the region scanner
                 if (pagingFilter.isStopped()) {
-                    // Close the current region scanner, start a new one and return a dummy result
-                    delegate.close();
-                    byte[] rowKey = pagingFilter.getRowKeyAtStop();
-                    boolean isInclusive = pagingFilter.isNextRowInclusive();
-                    scan.withStartRow(rowKey, isInclusive);
-                    delegate = region.getScanner(scan);
+                    byte[] rowKey = pagingFilter.getCurrentRowKeyToBeExcluded();
```

Comment:
move this to inside the `if (results.isEmpty())`

Reply:
Done